plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -496,6 +497,13 @@ public void set(PreparedStatement statement, int index, long value)
                 LocalDateTime date = LocalDateTime.from(Instant.ofEpochMilli(utcMillis).atZone(ZoneOffset.UTC));
                 statement.setString(index, DATE_FORMATTER.format(date));
             }
+
+            @Override
+            public void setNull(PreparedStatement statement, int index)
+                    throws SQLException
+            {
+                statement.setNull(index, Types.DATE);
```

Comment:
for values we use `statement.setString()` (a varchar), because we also you the bind expression `TO_DATE(?, 'SYYYY-MM-DD')`

i think this should be `Types.VARCHAR` here

Reply:
You are right. Fixed